includes/validation/class-amp-invalid-url-post-type.php

```diff
@@ -2130,4 +2131,46 @@ public static function filter_bulk_post_updated_messages( $messages, $bulk_count
 
 		return $messages;
 	}
+
+	/**
+	 * Is AMP Enabled on Post
+	 *
+	 * @param WP_Post $post              Post object to check.
+	 * @param array   $validation_errors Validation errors.
+	 * @param array   $counts            Error counts.
+	 *
+	 * @return bool|void
+	 */
+	public static function is_amp_enabled_on_post( $post, $validation_errors = array(), $counts = array() ) {
+		if ( empty( $post ) ) {
+			return;
+		}
+		if ( empty( $validation_errors ) ) {
+			$validation_errors = self::get_invalid_url_validation_errors( $post );
+		}
+		if ( empty( $counts ) ) {
+			$counts = array_fill_keys(
+				array( 'new_accepted', 'ack_accepted', 'new_rejected', 'ack_rejected' ),
+				0
+			);
+		}
+		foreach ( $validation_errors as $error ) {
+			switch ( $error['term']->term_group ) {
+				case AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_REJECTED_STATUS:
+					$counts['new_rejected']++;
+					break;
+				case AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_ACCEPTED_STATUS:
+					$counts['new_accepted']++;
+					break;
+				case AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_ACCEPTED_STATUS:
+					$counts['ack_accepted']++;
+					break;
+				case AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACK_REJECTED_STATUS:
+					$counts['ack_rejected']++;
+					break;
+			}
+		}
```

Comment:
This `foreach` loop is duplicated with `display_invalid_url_validation_error_counts_summary`. I suggest adding a `protected` method for `count_invalid_url_validation_errors()` which you pass the `$validation_errors` array both here and in `display_invalid_url_validation_error_counts_summary`.